.github/workflows/CI.yml

```
@@ -33,6 +57,7 @@ jobs:
 
   nodeJsBaselineAptCompatibility:
     name: NodeJS installed from stock Ubuntu-LTS packages (not external sources)
+    needs: [codeQuality]
```

@marcalexiei as stated above, this job should not depend on codeQuality either, it's the other way around

---

if there are lint or format errors I don't see why jobs that re running the build task.
/cc @escapedcat let me know if codeQuality should be ruined in parallel with other jobs

---

if the build fails, then the developer needs to fix the build first; nothing else matters more than that (lint or formatting errors of something that doesn't build are irrelevant, or even misleading)

---

@escapedcat let me know which approach do you prefer. I'm ok with both of them.

---

approach? man, this is similar to trying to run the tests before running the build! codeQuality needs to depend on a working build, not the other way around

---

I thought free minutes were limited, but that's only for private repositories.
My idea was: if code-quality is failing, the PR is not getting merged, and failing fast is cheaper (<1 minute rather than 10 minutes).

Since minutes are not an issue, we can do what you are suggesting.

---

I'm not proposing anything that would consume more minutes, I'm saying the dependencies were upside down, codeQuality needs to depend on build, but you were adding the dependency in the wrong direction

---

Thanks for your feedback @knocte .  I added a 👍 to that below. Now that I'm reading this let me add one more statement.

In most projects I know the tasks run independent of that to indicate fails in all tasks immediately. So even if the build fails you also see that linting fails and you can handle all the things instead of first fixing the build and then see that you also need to fix linting.
In the last project this was explicitly changed to give faster feedback.
I agree that with a non-working build a whole commit/PR is pointless, but faster feedback seems valuable?
So not sure what should wait for what and if tasks need to wait.
wdyt?
(Sorry if this is adding more confusion. But we're not in a rush and I hope this should be a somewhat fun project to work on and give people a chance to learn from each other)

---

> we're not in a rush

agreed

>  this should be a somewhat fun project

agreed

> but faster feedback seems valuable?

problem I see with this point of view is that a project that doesn't build could be causing cascading errors in other tools such as lint or formatting, so you might get **misleading** feedback, that's why waiting for the build to finish is worth it, instead of launching this job in parallel too

---

Yeah, I get your point. I guess it depends ™️ .  
@marcalexiei let's go with @knocte suggestion. We can always adjust it later on if there's a reason.